src/server/rdb_save.cc

```diff
@@ -952,18 +951,45 @@ void AppendFooter(io::StringSink* dump_res) {
    * RDB version and CRC are both in little endian.
    */
   const auto ver = MakeRdbVersion();
-  dump_res->Write(to_bytes(ver));
-  const auto crc = MakeCheckSum(dump_res->str());
-  dump_res->Write(to_bytes(crc));
+  auto ver_bytes = to_bytes(ver);
+  dump_res->Write(ver_bytes);
+
+  crc = crc64(crc, ver_bytes.data(), ver_bytes.size());
+  const auto crc_buf = WrapCheckSum(crc);
+  dump_res->Write(to_bytes(crc_buf));
 }
 }  // namespace
 
-void SerializerBase::DumpObject(const CompactObj& obj, io::StringSink* out) {
+void SerializerBase::DumpObject(const CompactObj& obj, io::Sink* out) {
   CompressionMode compression_mode = GetDefaultCompressionMode();
   if (compression_mode != CompressionMode::NONE) {
     compression_mode = CompressionMode::SINGLE_ENTRY;
   }
-  RdbSerializer serializer(compression_mode);
+  uint64_t crc = 0;
+  uint64_t total_size = 0;
+  auto flush = [&](RdbSerializer* srz) {
+    io::StringSink s;
+
```

Comment:
Do want to flush directly to a string and then to the sink ? Is there a way to do it directly ?

Reply:
the reason I do it as such is to be able to compute the CRC value for the additional data, so we can flush to the sink directly (otherwise we can't compute the crc)

Reply:
> (otherwise we can't compute the crc)

Yep that answers it :smile: